verify.go

```diff
@@ -153,10 +153,11 @@ func verifySignature(p7 *PKCS7, signer signerInfo, truststore *x509.CertPool) (e
 			}
 		}
 	}
-	if truststore != nil {
-		_, err = verifyCertChain(ee, p7.Certificates, truststore, signingTime)
+	if opts.Roots != nil {
+		opts.CurrentTime = signingTime
```

Comment:
If KeyUsage isn't set in the opts, we should maintain the same behavior and set it equal to ExtKeyUsageAny, otherwise it defaults to ServerAuth https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/crypto/x509/verify.go;l=810-812